### ISSUE TITLE
TCP services doesn't refresh backend endpoints.

### ISSUE THREAD

**Issue by donCappo** [haproxytech/kubernetes-ingress#155]
Hi,

after some fiddling, I was able to set-up tcp services for ingress controller.
However, the controller doesn't reflect the change in endpoints for corresponding svc it watches from the config-map.
It does reflect the change when adding new service in config-map . But not other way, i.e. scale up etc ...

**Comment by Mo3m3n** [comment]
Hi @donCappo 
Thanks you for reporting this bug!
This was fixed in this commit https://github.com/haproxytech/kubernetes-ingress/commit/1e782b53cbd2c3bb6eea433d05ca20e0a03e99b4. While waiting for the next release you can already test it with the dev image at `haproxytech/kubernetes-ingress:dev`

> after some fiddling, I was able to set-up tcp services for ingress controller.

Documentation of TCP services is available at the [controller doc](https://github.com/haproxytech/kubernetes-ingress/blob/master/documentation/controller.md). 
Let us know if this is not clear or not easy to find and we will update accordingly.

**Comment by donCappo** [comment]
Works like a charm! Thanks for quick fix.

On the fiddling side, well it's my fault not reading the docs more thoroughly.
You clearly state, inside the config map it has to be namespace/name format and for some reason I got lost inside all the configmaps we have. In the example you named config map "tcp" but it's inside default namespace and I reused the example incorrectly with serviceName/portName. My mistake.
Plus we have ingresses per namespace/project(openshift), so I was just setting it on namespace level.

Regarding this per namespace deployment - might be enhancement :) you still need a clusterRole for the serviceAccount, first I tried just the role for given namespace, which didn't work, "controller" probably expects to have cluster wide permissions.  I used the whitelisting for now, which works great, although I noticed that sometimes controller logs a message of attempt deleting service which happened in different namespace.
